### ISSUE TITLE
pri ploche to padne so zapornym indexom pri niektorych datach

### ISSUE THREAD

**Issue by RobertSpir** [xmatakt/bakalarka#5]
subor [http://tesla.math.sk/zem_sorted.7z](url)

**Comment by xmatakt** [comment]
Myslim/dufam ze to bude koli zmene progress baru pri pocitani normal, ked
je vela prilis vela dat, tusim to tam nasobim tromi a to uz prelezie int
takze hadze zaporne cisla ktore progress bar nema rad. Pri sfere som to
osetril pomocnym floatom a upravou poradia delenia a nasobenia, a zda sa mi
ze pri ploche som nechal staru verziu. Vecer sa na to pozriem.
Dňa 19.3.2016 10:23 používateľ "RobertSpir" notifications@github.com
napísal:

**Comment by RobertSpir** [comment]
moze byt, pada to na progressbare

**Comment by RobertSpir** [comment]
potom to skalovanie by bolo dobre spravit s ohladom na rozsah dat, pretoze ked som tam skusil toto
[http://tesla.math.sk/sl1.7z ](url), tak to aj pri 50 bolo cele veeeeelmi divoke

**Comment by RobertSpir** [comment]
nechcem byt hnidopich, ale to skalovanie je stale trochu blbe :D
pri zem_sorted je to aj pri 1 stale dost ploche, zato pri sl1 je to pri 1 zase moc divoke....
mozno by to mohlo byt tak, ze max "vyska" pri skale 1 by bola 50% sirky celej plochy, pri skale 50 by to bolo 1% sirky plochy...
a nie su tam nejake memory leaky? nejake opengl objekty, co by bolo treba manualne uvolnovat? Lebo ked som skusal tam na striedacku otvarat tie velke subory, tak potom uz spotreba pamate neklesla pod 2GB, aj ked som tam otvoril nejaky maly subor a dorobil tam gombik, ktory manualne vola GC.Collect();

**Comment by xmatakt** [comment]
O skalovani viem to som tam narychlo dal nieco co aspon trocha funguje a
memory leaky su celkom pravdepodobne, neviem ci este aj v tejto poslednej
ale v niektorych verziach som mal poznamky v komentoch. Tento vikend
neprebehol podla mojich predstav, iba vcera vecer som sa dostal k ciselnemu
popisu legendy a zistil som, ze napratat tam text nebude az take jednoduche
ako som si predstavoval, takze sa idem babrat s tym a vase poznatky ukladam
do fronty pre dalsie spracovanie.
Dňa 22.3.2016 19:12 používateľ "RobertSpir" notifications@github.com
napísal:

**Comment by RobertSpir** [comment]
cez velku noc sa skuste pozriet na ten volume rendering, ak budete mat cas. Tieto drobnosti ako memory leaky mozeme riesit aj na konci. Ak sa Vam na SVK podari ukazat softver s funkcnym volume renderingom, tak neverim tomu, aby to nebolo niektore z prvych troch miest...

**Comment by xmatakt** [comment]
Teraz u nas zavladla jarna stavebna horucka, ale urcite sa na to pozriem.
Ale ten text co sa pokusam pridat sa sprava uplne nelogicky a zatial neviem
preco, robim to tak, ze si vytvorim texturu s bitmapy na ktorej su potrebne
znaky ktora sa potom natiahne na trojuholniky ktorym zadam polohu na
obrazovke, akurat ze tam kde chcem je vzdy len prvy bod trojuholníka,
ostatne rozhadze ako sa mu zachce. Dokonca, ked definujem trojuholnik,
ktory ma vsetky vrcholy len v jednom bode, tak aj tak vytvori trojuholnik
cez pol obrazovky :D

cez velku noc sa skuste pozriet na ten volume rendering, ak budete mat cas.
Tieto drobnosti ako memory leaky mozeme riesit aj na konci. Ak sa Vam na
SVK podari ukazat softver s funkcnym volume renderingom, tak neverim tomu,
aby to nebolo niektore z prvych troch miest...

**Comment by xmatakt** [comment]
*z...
Inak na tu svocku treba mat aj nejaku pracu?, myslim nieco vo forme textu
Dňa 24.3.2016 13:34 používateľ "Timotej Maták" timotej.matak@gmail.com
napísal:

**Comment by RobertSpir** [comment]
Ano, ale to staci aj len tak 10 stran a je to az koncom aprila, takze si myslim, ze to bude v pohode...
v tej mojej verzii mam tu farebnu skalu spravenu cez Sprite na ktory nalepim texturu z bitmapy a potom len ten sprite umiestnim na obrazovku, bez nejakych trojuholnikov alebo niecoho. V opengl sa to vraj vola billboard...

**Comment by xmatakt** [comment]
Cital som si daco o tom, to je skor billboarding ako technika, nie je ziaden GL objekt billboard. Ide tam tusim o to, ako docielit, aby som mohol vyuzit 2D texturu na iluziu 3D objektu, napr. stromu, ze plocha na ktorej je textura vykreslena (zvacsa zlozena z trojuholnikov) je vzdy kolma na smer, ktorym sa kamera "pozera". Ale ja tom pouzivam ortogonalnu projekciu a specialne shadere pre text, takze nemusim riesit ten problem ktory riesi billboarding.
